pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -996,6 +996,13 @@ def __init__(
         self._padding_right = 2 ** len(wires) - 2 - self._padding_left
         self._CX = None
 
+        self._hyperparameters = {
+            "target_wire": self._target_wire,
+            "work_wires": self._work_wires,
+            "control_wires" : self._control_wires,
+            "control_values": self.control_values
+        }
```

Comment:
It still feels wrong to me that this has to be private. I think this is almost an argument for implementing a very simple setter for `self.hyperparameters`, so that you can do

```python
self.hyperparameters = {...}
```

We could also add some basic validation (check that the input is a dict, etc)

Reply:
We are setting it privately within the class. 

Once we create the dictionary during initialization, we can use `self.hyperparameters` instead.

Reply:
Right, but we shouldn't have to set a private attribute to set up an operation --- all developer facing APIs for creating an operator should be public, documented, and include validation where it makes sense